sds_data_manager/lambda_code/SDSCode/queries.py

```diff
@@ -28,14 +33,62 @@ def lambda_handler(event, context):
     logger.info(f"Context: {context}")
 
     logger.info("Received event: " + json.dumps(event, indent=2))
-    # TODO: Replace result using RDS
-    search_result = ""
-    logger.info("Query Search Results: " + json.dumps(search_result))
+
+    # add session, pick model like in indexer and add query to filter_as
+    query_params = event["queryStringParameters"]
+
+    # select the file catalog for the query
+    query = select(models.FileCatalog.__table__)
+    # get a list of all valid search parameters
+    valid_parameters = [
+        column.key
+        for column in models.FileCatalog.__table__.columns
+        if column.key not in ["id", "status_tracking_id"]
+    ]
+    # go through each query parameter to set up sqlalchemy query conditions
+    for param, value in query_params.items():
+        # confirm that the search parameter is valid
+        if param not in valid_parameters:
+            response = {
+                "statusCode": 400,
+                "body": json.dumps(
+                    f"{param} is not a valid search parameter. "
+                    + f"Valid search parameters are: {valid_parameters}"
+                ),
+                "headers": {
+                    "Content-Type": "application/json",
+                    "Access-Control-Allow-Origin": "*",  # Allow CORS
+                },
+            }
+            return response
+        # check if we're search for start_date or end date to
+        # setup the correct "where" time condition
+        if param == "start_date":
+            query = query.where(
+                models.FileCatalog.start_date
+                >= datetime.datetime.strptime(value, "%Y%m%d")
+            )
+        elif param == "end_date":
+            # TODO: Need to discuss as a team how to handle date queries. For now,
+            # the date queries will only look at the file start_date.
+            query = query.where(
+                models.FileCatalog.start_date
+                <= datetime.datetime.strptime(value, "%Y%m%d")
+            )
+        # all non-time string matching parameters
+        else:
+            query = query.where(getattr(models.FileCatalog, param) == value)
```

Comment:
Out of curiosity and knowing nothing about this, is this the standard way of combining the query conditions, or is it faster to do something like combining all the conditions together into a single `where` clause? `query.where(logical_condition1 and logical_condition2 and logical_condition3 ...)`

Reply:
I'm not sure what the standard way is, but I saw both in documentation. Appending wheres like this seemed to me like the best way to add all the query parameters while handling the dates.

Reply:
I'm not sure what the difference would be as far as speed, but from my understanding, the query variable is converted to sql when it's executed, so I'm not sure if the resulting sql would be different

Reply:
:+1: I agree this seems pretty clean and the documentation indicates that the where returns a new statement connected by an AND, so as long as thats what you want this seems reasonable.
https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.Select.where
> Return a new [select()](https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.select) construct with the given expression added to its WHERE clause, joined to the existing clause via AND, if any.

But, if you are curious, it looks like you could compare the two results and see if they differ at all:
https://stackoverflow.com/questions/5631078/sqlalchemy-print-the-actual-query